File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Session/SessionDto.java
package com.salesianos.triana.BeElite.dto.Session;

import com.salesianos.triana.BeElite.model.Session;
import lombok.Builder;

import java.time.DayOfWeek;
import java.util.List;

@Builder
public record SessionDto(String date,
                         Long sessionNumber,
                         int sameDaySessionNumber)
{
    public static SessionDto empty(){
        return SessionDto.builder().build();
    }
    public static SessionDto of(Session s){
        return SessionDto.builder()
                .date(s.getDate().toString())
                .sessionNumber(s.getId().getSession_number())
                .sameDaySessionNumber(s.getSame_day_session_number())
                .build();
    }
}

File: frontend/be_elite/lib/models/Session/session_dto.dart
class SessionDto {
  String? date;
  int? sessionNumber;
  int? sameDaySessionNumber;

  SessionDto({this.date, this.sessionNumber, this.sameDaySessionNumber});

  factory SessionDto.fromJson(Map<String, dynamic> json) => SessionDto(
      date: json['date'] as String?,
      sessionNumber: json['sessionNumber'] as int?,
      sameDaySessionNumber: json['sameDaySessionNumber'] as int?);

  Map<String, dynamic> toJson() => {
        'date': date,
        'sessionNumber': sessionNumber,
        'sameDaySessionNumber': sameDaySessionNumber,
      };
}
